x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/common/api/util/with_existing_migration_id.ts

```diff
@@ -26,9 +28,13 @@ export const withExistingMigration = <
 ): RequestHandler<P, Q, B, SecuritySolutionRequestHandlerContext, Method> => {
   return async (context, req, res) => {
     const { migration_id: migrationId } = req.params;
+    const pathParts = req.route.path.split('/');
     const ctx = await context.resolve(['securitySolution']);
-    const ruleMigrationsClient = ctx.securitySolution.siemMigrations.getRulesClient();
-    const storedMigration = await ruleMigrationsClient.data.migrations.get(migrationId);
+    const migrationsClient: SiemRuleMigrationsClient | SiemDashboardMigrationsClient =
+      pathParts.includes('rules')
```

Comment:
This looks good. Can you please remove `with_existing_dashboard_migration` and its references as well. 

And add tests for both the cases.